dashboard/app/models/script.rb

```diff
@@ -227,19 +228,27 @@ def self.cache_find_script_level(script_level_id)
     script_level
   end
 
-  # Find the level with the given id from the cache, unless the level build mode
-  # is enabled in which case it is always fetched from the database. If we need to fetch
-  # the level and we're not in level mode (for example because the level was created after
-  # the cache), then an entry for the level is added to the cache.
-  def self.cache_find_level(level_id)
-    level = level_cache[level_id] if self.should_cache?
+  # Find the level with the given id or name from the cache, unless the level
+  # build mode is enabled in which case it is always fetched from the database.
+  # If we need to fetch the level and we're not in level mode (for example
+  # because the level was created after the cache), then an entry for the level
+  # is added to the cache.
+  # @param level_identifier [Integer | String] the level ID or level name to
+  #   fetch
+  # @return [Level] the (possibly cached) level
+  def self.cache_find_level(level_identifier)
+    level = level_cache[level_identifier] if self.should_cache?
+    return level unless level.nil?
 
     # If the cache missed or we're in levelbuilder mode, fetch the level from the db.
-    if level.nil?
-      level = Level.find(level_id)
-      # Cache the level, unless it wasn't found.
-      @@level_cache[level_id] = level if level && self.should_cache?
-    end
+    level = if level_identifier.is_a? Integer
+              Level.find(level_identifier)
+            else
+              Level.find_by_name(level_identifier)
```

Comment:
Your note caused me to preserve existing behavior when the level ID cannot be found. Rather than returning `nil`, the method raises an exception.